storage/store.go

```diff
@@ -493,20 +494,26 @@ func (s *Store) Start(stopper *util.Stopper) error {
 	return nil
 }
 
+// WaitForInit waits for any asynchronous processes begun in Start()
+// to complete their initialization. In particular, this includes
+// gossiping. In some cases this may block until the range GC queue
+// has completed its scan. Only for testing.
+func (s *Store) WaitForInit() {
+	s.initComplete.Wait()
+}
+
 // startGossip runs an infinite loop in a goroutine which regularly checks
 // whether the store has a first range or config replica and asks those ranges
 // to gossip accordingly.
 func (s *Store) startGossip() error {
-	// Go through one iteration synchronously before returning. This makes sure
-	// that everything is gossiped when the store finishes starting.
-	if err := s.maybeGossipConfigs(); err != nil {
-		return err
-	}
-	if err := s.maybeGossipFirstRange(); err != nil {
-		return err
-	}
+	s.initComplete.Add(2)
```

Comment:
But that's different from what we have here: the call to Done() is in the middle of the function. I can change it into two calls to Add(1), each before their respective RunWorker, but I think that abstracting this further would be counterproductive (unless and until we have more than two of these)